src/main/java/org/fungover/Topic.java

```diff
@@ -0,0 +1,68 @@
+package org.fungover;
+
+import java.util.Objects;
+
+public class Topic {
+    private String name;
```

Comment:
Using 'final' will make the 'name' variable immutable after it has been assigned a value. But I'm not sure if making it final is the best choice here. The 'name' might require changes during the object's lifetime.

Reply:
I have no idea how Topics work other than a quick look at the documents from Martin 😅 But I do think declaring a variable final is a good practice until the situation comes when the variable needs to change, and then the class is updated to handle that. You could implement a setName() so that it is possible to change the topic objects name if needed. 👍

Reply:
Implementing a setter is a good point, that makes the topic 'name' still mutable 👍🏽 Thanks!

Reply:
Do we really need to change Topic name? The topic class is more about representing a topic that a client is subscribed to or a topic that a client want to send a message to. Instead of using strings in our code we can use objects of type Topic that internally stores a String representing our topic.

Using Topic datatypes in the code gives us some help from the compiler with checking that we never can create a topic object that is invalid and then all other functions in our program can assume that when receiving a Topic object we don't need to verify it again.

Therefor I suggest implementing the Topic class as an immutable class so all places holding a reference to an object of type Topic knows that it can't change. If we need to do changes we can always create new objects.

Using immutable objects here also gives us the possibility of reusing Topic objects for all clients registering for messages on same topic name.

Reply:
However, from my perspective, it appears that the `name` variable is never mutated; it is merely assigned within the Topic creation method. If there is a concern regarding mutation, a preferable alternative might be the use of [StringBuffer ](https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuffer.html)or [StringBuilder](https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html).

However, as previously mentioned, I do not observe any mutation in your code. Therefore, the use of `final` seems appropriate to me as well